src/v/iceberg/tests/schema_protobuf_test.cc

```diff
@@ -0,0 +1,245 @@
+        IsField(1, "id", int_type{}), IsField(2, "name", string_type{})));
+}
+
+/**
+ *enum State {
+ *  Active = 0;
+ *  InActive = 1;
+ *}
+ *
+ *message BrokerShard {
+ *  int64 id = 1;
+ *  int32 shard = 2;
+ *}
+ *
+ *message Partition {
+ *    int32 id = 1;
+ *    repeated BrokerShard replicas = 2;
+ *    State state = 3;
+ *}
+ *
+ *message Topic {
+ *    string name = 1;
+ *    repeated Partition partitions = 2;
+ *}
+ *
+ *message Metadata {
+ *    map<string,Topic> topics = 1;
+ *    repeated int64 nodes = 2;
+ *}
+ */
+// this schema contains a list map and nested fields
+// static_assert(
+// SchemaConverter<protobuf_schema_converter, google::protobuf::Descriptor>);
+TEST(SchemaProtobuf, TestComplexSchema) {
+    google::protobuf::DescriptorPool d_pool;
+    ASSERT_TRUE(parse_proto_definition_string(complex_proto_str, d_pool));
+    auto md = d_pool.FindMessageTypeByName("Metadata");
+
+    auto result = iceberg::type_to_iceberg(*md);
+    ASSERT_FALSE(result.has_exception());
+
+    auto& top_level = result.value();
+    auto& topics_map_field = top_level.fields[0];
+    auto& topics_map_type = std::get<map_type>(top_level.fields[0]->type);
+    EXPECT_EQ(topics_map_field->id, 1);
+    EXPECT_EQ(topics_map_field->name, "topics");
+    EXPECT_THAT(topics_map_type.key_field, IsField(1, "key", string_type{}));
+
+    EXPECT_EQ(topics_map_type.value_field->id, 2);
+    EXPECT_EQ(topics_map_type.value_field->name, "value");
+
+    auto& topic_type = std::get<struct_type>(topics_map_type.value_field->type);
+    EXPECT_THAT(topic_type.fields[0], IsField(1, "name", string_type{}));
+
+    EXPECT_EQ(topic_type.fields[1]->id, 2);
+    EXPECT_EQ(topic_type.fields[1]->name, "partitions");
+    auto& partition_list_type = std::get<list_type>(topic_type.fields[1]->type);
+    EXPECT_EQ(partition_list_type.element_field->id, 2);
+    EXPECT_EQ(partition_list_type.element_field->name, "element");
+
+    auto& partition_type = std::get<struct_type>(
+      partition_list_type.element_field->type);
+    EXPECT_THAT(partition_type.fields[0], IsField(1, "id", int_type{}));
+
+    EXPECT_EQ(partition_type.fields[1]->id, 2);
+    EXPECT_EQ(partition_type.fields[1]->name, "replicas");
+    auto& replica_list_type = std::get<list_type>(
+      partition_type.fields[1]->type);
+
+    // enum is mapped to signed integer
+    EXPECT_THAT(partition_type.fields[2], IsField(3, "state", int_type{}));
+
+    auto& broker_shard_type = std::get<struct_type>(
+      replica_list_type.element_field->type);
+    EXPECT_THAT(
+      broker_shard_type.fields,
+      ElementsAre(
+        IsField(1, "id", long_type{}), IsField(2, "shard", int_type{})));
+}
+
+TEST_CORO(SchemaProtobuf, TestProtoTestMessages) {
+    auto desc_path = test_utils::get_runfile_path(
+      "src/v/iceberg/tests/testdata/"
+      "iceberg_ready_test_messages_edition2023.proto");
+
+    if (!desc_path.has_value()) {
+        desc_path = "iceberg_ready_test_messages_edition2023.proto";
```

Comment:
should this acquired via `getenv` or something like that for the non-bazel case?